preprocessors/object-detection-azure/objdetect.py

```diff
@@ -0,0 +1,190 @@
+                      "type": result['objects'][item]['object'],
+                      "dimensions": normalised_dims,
+                      "confidence": class_prob,
+                      "centroid": centre,
+                      "area": area
+                      }
+        send.append(dictionary)
+    return send
+
+
+def process_image(image):
+
+    region = "canadacentral"  # For example, "westus"
+    api_key = os.environ["AZURE_API_KEY"]
+
+    # Set request headers
+    headers = dict()
+    headers['Ocp-Apim-Subscription-Key'] = api_key
+    headers['Content-Type'] = 'application/octet-stream'
+
+    # Set request querystring parameters
+    """Only query categories"""
+    params = {'visualFeatures': 'Objects'}
+
+    # Make request and process response
+    response = requests.request(
+        'post',
+        "https://{}.api.cognitive.microsoft.com/vision/v1.0/analyze".format(region),
+        data=image,
+        headers=headers,
+        params=params
+    )
+    if response.status_code == 200 or response.status_code == 201:
+
+        if 'content-length' in response.headers and \
+                int(response.headers['content-length']) == 0:
+            return "", 204
+        elif 'content-type' in response.headers and \
+                isinstance(response.headers['content-type'], str):
+            if 'application/json' in response.headers['content-type'].lower():
+                if response.content:
+                    result = response.json()
+                    label = process_results(result)
+                else:
+                    return "", 204
+            else:
+                return "", 204
+
+    else:
+        return "", 204
+
+    return label
+
+
+
+@app.route("/preprocessor", methods=['POST', ])
+def categorise():
+    logging.debug("Received request")
+    # load the schema
+    with open('./schemas/preprocessors/object-detection.schema.json') \
+            as jsonfile:
+        data_schema = json.load(jsonfile)
+    with open('./schemas/preprocessor-response.schema.json') \
+            as jsonfile:
+        schema = json.load(jsonfile)
+    with open('./schemas/definitions.json') as jsonfile:
+        definitionSchema = json.load(jsonfile)
+    with open('./schemas/request.schema.json') as jsonfile:
+        first_schema = json.load(jsonfile)
+    schema_store = {
+        schema['$id']: schema,
+        definitionSchema['$id']: definitionSchema
+    }
+    resolver = jsonschema.RefResolver.from_schema(
+        schema, store=schema_store)
+
+    content = request.get_json()
+    try:
+        validator = jsonschema.Draft7Validator(first_schema, resolver=resolver)
+        validator.validate(content)
+    except jsonschema.exceptions.ValidationError as e:
+        logging.error(e)
+        return jsonify("Invalid Preprocessor JSON format"), 400
+    request_uuid = content["request_uuid"]
+    timestamp = time.time()
+    name = "ca.mcgill.a11y.image.preprocessor.objectDetectionAzure"
```

Comment:
Wrong name. Should be objectDetection.

Reply:
Isn't that the name for the YOLO model?

Reply:
For the CI CD, please share documentation of how to make one.